graphql_compiler/cost_estimation/analysis.py

```diff
@@ -53,9 +54,81 @@ def _convert_int_interval_to_field_value_interval(
     return Interval(lower_bound, upper_bound)
 
 
+def get_single_field_filters(
+    schema_info: QueryPlanningSchemaInfo, query_metadata: QueryMetadataTable,
+) -> Dict[PropertyPath, Set[FilterInfo]]:
+    """Find the single field filters for each field.
+
+    Filters that apply to multiple fields, like name_or_alias, are ignored.
+    Filters inside fold scopes are not considered.
+
+    Args:
+        schema_info: QueryPlanningSchemaInfo
+        query_metadata: info on locations, inputs, outputs, and tags in the query
+
+    Returns:
+        A set of filters for each field.
```

Comment:
nit: There won't be a key-value pair if the field doesn't have any filters.

```suggestion
        dict mapping fields to their set of filters.
```